vault/resource_kmip_secret_role_test.go

```diff
@@ -0,0 +1,213 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccKMIPSecretRole_basic(t *testing.T) {
+	path := acctest.RandomWithPrefix("tf-test-kmip")
+	resourceName := "vault_kmip_secret_role.test"
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestEntPreCheck(t) },
+		CheckDestroy: testAccKMIPSecretRoleCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testKMIPSecretRole_initialConfig(path),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "path", path),
+					resource.TestCheckResourceAttr(resourceName, "scope", "scope-1"),
+					resource.TestCheckResourceAttr(resourceName, "role", "test"),
+					resource.TestCheckResourceAttr(resourceName, "tls_client_key_type", "ec"),
+					resource.TestCheckResourceAttr(resourceName, "tls_client_key_bits", "256"),
+					resource.TestCheckResourceAttr(resourceName, "operation_activate", "true"),
```

Comment:
Maybe update to use your shiny new constants instead?

Reply:
Updated in ab39eb6